client/api.py

```diff
@@ -5,25 +5,52 @@
 from flask import Flask
 from flask import request
 from flask.ext.cors import CORS
+from flask_limiter import Limiter
+from flask_limiter.util import get_remote_address
+
 
 import api_handler
 import blacklist
 from cache.connection import RedisInstance as Redis
+import cache_utils
+import constants
+from constants import StatusCode
 from config import config
 from query_utils import QueryParams
 
 app = Flask(__name__)
 CORS(app)
 
+limiter = Limiter(
+    app,
+    key_func=get_remote_address,
+    storage_uri="redis://%s:%s" % (config.REDIS_HOST, config.REDIS_PORT),
+    global_limits=config.GLOBAL_RATE_LIMIT
+)
+
+
+@limiter.request_filter
+def ip_whitelist():
+    return request.remote_addr == constants.LOCALHOST
+
+
+@app.errorhandler(StatusCode.RATE_LIMIT)
+def ratelimit_handler(e):
+    return api_handler.generate_rate_limit_response()
+
 
 @app.route('/')
+@limiter.exempt
 def home_page():
-    return "", 200
+    return "", StatusCode.OK
 
 
 @app.route('/healthcheck')
+@limiter.exempt
 def health_check():
-    return "Success!", 200
+    if cache_utils.is_redis_available():
```

Comment:
yeah but we need redis now because we're doing rate limiting with redis. it's no longer optional